content/blog/2024/anchor-position-yearbook.md

```diff
@@ -0,0 +1,181 @@
+name inside of the same `<li>`.
+
+```html
+<ul>
+  <li>
+    <img src="photo1.jpg" />
+    <h2>Name 1</h2>
+  </li>
+  <li>
+    <img src="photo2.jpg" />
+    <h2>Name 2</h2>
+  </li>
+  ...
+</ul>
+```
+
+We'll make the `<ul>` the main photo area, use `flex-wrap: wrap` to fit as many
+photos in a row as possible, and make it an anchor by giving it `anchor-name:
+--photos`.
+
+Then we can place the `<h2>` titles to meet the first constraint.
+
+```css
+h2 {
+  position: absolute;
+  left: calc(anchor(--photos right) + 1em);
+}
+```
+
+That's right -- we don't need to first figure out how many pixels from the left
+viewport edge the photos are, we can just declaratively describe what we want.
+"Place the left edge at the sum of photos' right side and `1em`".
+
+We can then satisfy the second constraint, that each name should be in the same
+row as its photo.
+
+```css
+li {
+  anchor-scope: --photo;
+}
+h2 {
+  top: anchor(--photo top);
+}
+```
+
+The `top` rule simply aligns the top of the name with the top of the photo. The
+`anchor-scope` is important here. By default, if there are multiple elements
+that have the same `anchor-name`, the last element in the DOM will be selected.
+Setting `anchor-scope` on a parent element means that the anchored element won't
+look outside of the parent element for an anchor. This means the `<h2>` will be
+aligned with the top of its photo.
+
+But wait -- we have one more constraint. At this point, all the names in a row are
+layered on top of each other. We want them to vertically stack instead.
+
+```css
+h2 {
+  anchor-name: --title;
+  top: max(anchor(--photo top), anchor(--title bottom, 0));
+}
+```
+
+Now, the top of the name will be either the top of the photo OR the bottom of
+the title above it, whichever is greater (or further down the page). That's
+because the anchoring element has to come **before** the anchored element, which
+means the reference to `anchor(--title)` will be the name above it.
+
+I'd recommend [viewing this on Codepen](https://codepen.io/jamessw/pen/BaXqgWd)
+so you can resize the demo. I've added a short transition to the `top` property
+so you can see how the names adjust as the names above them move.
+
+{{ embed.codepen(
+  id='BaXqgWd',
+  title='CSS Anchor Positioning - Yearbook layout',
+  user='jamessw'
+) }}
+
+As an added bonus, because the photo and name are in the same `<li>`, we can add
+hover effects so that if one is hovered, both parts are hovered. If the item is
+a link they can be inside the same `<a>`.
+
+## Wrap-up
+
+What new possibilities for anchor positioning are you excited about? Let us know
+by sending us a message on [Mastodon](https://front-end.social/@oddbird) or
+[BlueSky](https://bsky.app/profile/oddbird.dev).
+
+And if you're excited to use anchor positioning before it becomes baseline,
+check out our [polyfill](https://github.com/oddbird/css-anchor-positioning). It
+supports enough of the anchor positioning spec to do most things.
```

Comment:
I think it's worth some discussion of why the demo doesn't use the polyfill. That was confusing to me.

Reply:
I updated it to include the polyfill, and also added a note on anchor-scope not being polyfilled or present in Chromium 125-130.